src/main/java/org/mule/module/apikit/validation/body/form/transformation/MultipartFormDataBinaryParameter.java

```diff
@@ -56,7 +61,86 @@ public void validate(Parameter parameter) throws InvalidFormParameterException {
     }
   }
 
-  private boolean anyFileTypeAllowed(Set<String> fileTypes) {
-    return fileTypes.size() == 1 && fileTypes.contains("*/*");
+  private void validateMediaType(Set<String> acceptedMediaTypes) throws InvalidFormParameterException {
+    if (acceptedMediaTypes == null || acceptedMediaTypes.isEmpty()) {
+      return;
+    }
+
+    // If we support anything
+    if (acceptedMediaTypes.contains("*/*")) {
+      return;
+    }
+
+    // If we have an exact match
+    if (acceptedMediaTypes.contains(mediaType.toString())) {
+      return;
+    }
+
+    // If any media type is compatible
+    if (acceptedMediaTypes.stream().anyMatch(accepted -> isCompatible(accepted, mediaType))) {
+      return;
+    }
+
+    throw new InvalidFormParameterException(format("Invalid content type: %s", mediaType));
+  }
+
+  private static boolean isCompatible(String expected, MediaType given) {
+    try {
+      MimeType acceptedMimeType = new MimeType(expected);
+      return isCompatible(acceptedMimeType, given);
+    } catch (MimeTypeParseException e) {
+      return false;
+    }
+  }
+
+  private static boolean isCompatible(MimeType expected, MediaType given) {
+    String expectedPrimary = expected.getPrimaryType();
+    String expectedSub = expected.getSubType();
+    String givenPrimary = given.getPrimaryType();
+    String givenSub = given.getSubType();
+
+    // We treat all parameters in `expected` as required to be equal. Extra parameters on `give` are ignored.
+    MimeTypeParameterList parameters = expected.getParameters();
+    Enumeration<String> parameterNames = (Enumeration<String>) parameters.getNames();
+    while (parameterNames.hasMoreElements()) {
+      String name = parameterNames.nextElement();
+      String expectedValue = parameters.get(name);
+
+      // mule-api has a special case for when a parameter is called "charset"
+      if ("charset".equalsIgnoreCase(name)) {
+        if (!isSameCharsetOrAlias(expectedValue, given.getCharset().orElse(null))) {
+          return false;
+        }
+      } else {
+        if (!Objects.equals(expectedValue, given.getParameter(name))) {
+          return false;
+        }
+      }
+    }
+
+    // If we have the ANY media type then it's compatible
+    if (Objects.equals("*", expectedPrimary) && Objects.equals("*", expectedSub)) {
```

Comment:
is the same condition ? `    // If we support anything
    if (acceptedMediaTypes.contains("*/*")) {
      return;
    }`

Reply:
Kinda. This supports things like `*/*; charset=utf8`.